modules/ROOT/pages/values-and-types/type-predicate.adoc

```diff
@@ -0,0 +1,303 @@
+[[type-predicate]]
+= Type predicate expression
+
+_This feature was introduced in Neo4j 5.9._
+
+A type predicate expression can be used to verify the type of a variable, literal, property or other Cypher expression.
+
+[[type-predicate-syntax]]
+== Syntax
+
+[source, syntax]
+----
+<expr> IS :: <TYPE>
+----
+
+Where `<expr>` is any Cypher expression and `<TYPE>` is a Cypher type.
+For all available Cypher types, see the section on xref::values-and-types/property-structural-constructed.adoc#types-synonyms[types and their synonyms].
+
+[[type-predicate-regular]]
+== Verify the type of a Cypher expression
+
+[source, cypher]
+----
+UNWIND [42, true, 'abc'] AS val
+RETURN val, val IS :: INTEGER AS isInteger
+----
+
+[role="queryresult",options="header,footer",cols="2*<m"]
+|===
+| val | isInteger
+| 42 | true
+| true | false
+| 'abc' | false
+2+d|Rows: 3
+|===
+
+[[type-predicate-not]]
+== Type predicate expressions with NOT
+
+It is also possible to verify that a Cypher expression is not of a certain type, using the negated type predicate expression `IS NOT ::`.
+
+[source, cypher]
+----
+UNWIND [42, true, 'abc'] AS val
+RETURN val, val IS NOT :: STRING AS notString
+----
+
+[role="queryresult",options="header,footer",cols="2*<m"]
+|===
+| val | notString
+| 42 | true
+| true | true
+| 'abc' | false
+2+d|Rows: 3
+|===
+
+[[type-predicate-null]]
+== Type predicate expressions for null
+
+As of Neo4j 5.10, a Cypher type includes the `null` value unless it is explicitly appended with `NOT NULL`.
```

Comment:
Actually, in 5.9 it also included the null value, what is new is the ability to remove it using that syntax